.devcontainer/Dockerfile

```diff
@@ -30,26 +33,40 @@ COPY .devcontainer/scripts/docker-client.sh /tmp/
 RUN /tmp/docker-client.sh $USERNAME
 
 # Install Docker
-RUN apt-get update && sudo apt-get install -y  apt-transport-https ca-certificates curl gnupg  lsb-release \
+RUN apt-get update && sudo apt-get install -y apt-transport-https ca-certificates curl gnupg lsb-release \
     && curl -fsSL https://download.docker.com/linux/debian/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
     && echo "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/debian $(lsb_release -cs) stable" \
     | tee /etc/apt/sources.list.d/docker.list > /dev/null \
-    && apt-get update && apt-get install -y docker-ce docker-ce-cli containerd.io
+    && apt-get update && apt-get install -y docker-ce docker-ce-cli containerd.io \
+    && apt-get clean -y && rm -rf /var/lib/apt/lists/*
 
 # Install nekos act - run GitHub workflows locally https://github.com/nektos/act
-RUN curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash
+RUN if [ "${INTERACTIVE}" = "true" ]; then curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash; fi
 
 # Install Certbot
-RUN apt-get install -y python3 python3-venv libaugeas0 \
+RUN apt-get update && apt-get install -y python3 python3-venv libaugeas0 \
     && python3 -m venv /opt/certbot/ \
-    && /opt/certbot/bin/pip install --upgrade pip \
-    && /opt/certbot/bin/pip install certbot
+    && /opt/certbot/bin/pip install --no-cache-dir --upgrade pip \
+    && /opt/certbot/bin/pip install --no-cache-dir certbot \
+    && apt-get clean -y && rm -rf /var/lib/apt/lists/*
 
 # Install Porter
-RUN export PORTER_HOME=/home/$USERNAME/.porter \
-    && curl -L https://cdn.porter.sh/latest/install-linux.sh | bash \
-    && ${PORTER_HOME}/porter mixin install docker \
-    && chown -R $USERNAME ${PORTER_HOME}
+# Not using the script from https://cdn.porter.sh/latest/install-linux.sh
+# as it installes things we don't need and duplicates the binary.
+ARG PORTER_MIRROR=https://cdn.porter.sh
+ARG PORTER_PERMALINK=v0.38.10
```

Comment:
Real petty and feel free to ignore - do you think this would be better in a scripts\porter.sh

(just a thought!)

Reply:
done